src/isl/util.rs

```diff
@@ -35,210 +35,147 @@ impl Range {
         use RangeBoundaryValueType::*;
         match self {
             Range::Integer(start, end) => {
-                match (start, end) {
-                    (Min, Max) => {
-                        unreachable!("Range boundaries can not be min and max together (i.e. range::[min, max] is not allowed)")
-                    },
-                    (Min, Value(end_value, boundary_type)) => {
-                        match end_value {
-                            Integer(range_end_value) => {
-                                return match range_end_value {
-                                    AnyInt::I64(int_end_value) => {
-                                        Ok(int_end_value > &value.as_any_int().unwrap().as_i64().unwrap() || (*boundary_type == RangeBoundaryType::Inclusive && int_end_value == &value.as_any_int().unwrap().as_i64().unwrap()))
-                                    }
-                                    AnyInt::BigInt(big_int_end_value) => {
-                                        Ok(big_int_end_value > &value.as_any_int().unwrap().as_big_int().unwrap() || (*boundary_type == RangeBoundaryType::Inclusive && big_int_end_value == value.as_any_int().unwrap().as_big_int().unwrap()))
-                                    }
-                                }
-                            },
-                            _ => unreachable!("Integer range can only have integers as lower and upper range boundary value")
-                        }
-                    },
-                    (Value(start_value, boundary_type), Max) => {
-                        match start_value {
-                            Integer(range_start_value) => {
-                                return match range_start_value {
-                                    AnyInt::I64(int_start_value) => {
-                                        Ok(int_start_value < &value.as_any_int().unwrap().as_i64().unwrap() || (*boundary_type == RangeBoundaryType::Inclusive && int_start_value == &value.as_any_int().unwrap().as_i64().unwrap()))
-                                    }
-                                    AnyInt::BigInt(big_int_start_value) => {
-                                        Ok(big_int_start_value < &value.as_any_int().unwrap().as_big_int().unwrap() || (*boundary_type == RangeBoundaryType::Inclusive && big_int_start_value == value.as_any_int().unwrap().as_big_int().unwrap()))
-                                    }
-                                }
-                            },
-                            _ => unreachable!("Integer range can only have integers as lower and upper range boundary value")
-                        }
+                let unwrapped_value = value.as_any_int().unwrap();
```

Comment:
A few thoughts here:

1. If this is a `Range<integer>` and I ask whether a `float` is in the range, I think this should return an `Err` instead of panicking.
2. I think eventually we'll need an enum variant for `Range<number>` [as described here](https://amzn.github.io/ion-schema/docs/spec.html#rangenumber). This is where we'd convert any/all types to `decimal` during comparison. We can do this in a follow-on PR.
3. (Just an observation, no change needed.) We don't have any use for `value` after it gets unwrapped. This seems like a situation where variable shadowing would work nicely.

```rust
let value = value.as_any_int().unwrap();
```

This creates a new variable with the same name (`value`) but a different type (`AnyInt` instead of `Option<AnyInt>`).

Reply:
For 2, I meant to create an issue to track it. Here's the reference issue: https://github.com/amzn/ion-schema-rust/issues/58